`lastArg` is set to undefined when last argument is `false`

**Describe the bug**
When using fakes, `lastArg` is set to `undefined` whenever the last argument passed is `false`.

The primary issue is in `fake.js#wrapFunc` where the assignment is using `&&` -
```js
var lastArg = (arguments.length > 0 && arguments[arguments.length - 1]) || undefined;
```
`lastArg` will always be `undefined` when the last passed argument is `false` not just _falsy_.

**To Reproduce**
Make an expectation of the last argument
_example uses `chai`_
```js
const foo = {
  greet() {
    this.speak("hello, world", false);
  },
  speak(msg, speakTwice) {
    console.log(msg);
    if (speakTwice) {
      console.log(msg);
    }
  }
};

const fake = sinon.fake();
sinon.replace(foo, "speak", fake);

foo.greet();

expect(fake.called).to.equal(true);
expect(fake.lastCall.lastArg).to.equal(false);
// => AssertionError: expected undefined to equal false
```

**Expected behavior**
`fake.lastCall.lastArg` should return `false` when the last argument of a function call is actually `false`.

**Context (please complete the following information):**

* Library version: `7.2.5`
* Environment: `nodejs@10.15` `npm@6.4.1` on `macOS 10.14`
* Other libraries you are using:
```json
"chai": "^4.2.0",
"mocha": "^5.2.0",
"sinon": "^7.2.5",
"ts-node": "^8.0.2",
```

--- a/lib/sinon/fake.js
+++ b/lib/sinon/fake.js
@@ -37,7 +37,7 @@ function cleanProxy(f) {
 var uuid = 0;
 function wrapFunc(f) {
     var fakeInstance = function() {
-        var lastArg = (arguments.length > 0 && arguments[arguments.length - 1]) || undefined;
+        var lastArg = arguments.length > 0 ? arguments[arguments.length - 1] : undefined;
         var callback = lastArg && typeof lastArg === "function" ? lastArg : undefined;
 
         /* eslint-disable no-use-before-define */

--- a/test/fake-test.js
+++ b/test/fake-test.js
@@ -143,6 +143,18 @@ describe("fake", function() {
             f(46);
             assert.equals(f.lastArg, 46);
 
+            f(false, true, 47, "string");
+            assert.equals(f.lastArg, "string");
+
+            f("string", false, true, 47);
+            assert.equals(f.lastArg, 47);
+
+            f(47, "string", false, true);
+            assert.equals(f.lastArg, true);
+
+            f(true, 47, "string", false);
+            assert.equals(f.lastArg, false);
+
             f();
             refute.defined(f.lastArg);
         });

--- a/test/issues/issues-test.js
+++ b/test/issues/issues-test.js
@@ -588,4 +588,14 @@ describe("issues", function() {
             assert.equals(stub({}, [], "b"), "b");
         });
     });
+
+    describe("#1986", function() {
+        it("should not set `lastArg` to undefined when last argument is `false`", function() {
+            var fake = sinon.fake();
+
+            fake(99, false);
+
+            assert.equals(fake.lastArg, false);
+        });
+    });
 });

